### DISCUSSION TITLE
Building a Regulated Asset with Stellar's Soroban

### DISCUSSION THREAD

**Discussion by fazzatti** [stellar/sorobounty-spectacular#12] [General]
Hi! Fifo here! 👋 

I've been working on a website to expand on my youtube channel and inspired by the sorobounty spectacular, I decided to initiate a video series around a soroban use case. I'll continue to expand on this use case by optimizing and adding more features as the series continues. In the future, I plan to also work on other use cases and material to go alongside.

The use case I chose is inspired by SEP08 and showcases how we can leverage programmability to improve several aspects of that standard. I reckon the series will probably continue for quite some time as I'm full of ideas and next steps to tackle but I believe it is a good starting point and I would love to get feedback to help me steer this kind of content in the right direction.

Here are the links:

- The main repo: [Oi Fifo Soroban Demos](https://github.com/fazzatti/oi-fifo-soroban-demos/blob/e8f684e122b9d2fa60e4ee718706202e1a60bd89/README.md)
- The hosted demo: [oififo.com/demos/regulated-asset](https://www.oififo.com/demos/regulated-asset/)
- The youtube series: [🚀 Exploring Regulated Assets with Stellar's Soroban](https://youtube.com/playlist?list=PLJo7htkGqBrFIHg6keCRRYjf6AEB1m1wo&si=ACexy42d8nU6mOeD)

Hope you guys like it!

**Comment by briwylde08** [comment]
Hello! Thank you so much for your Sorobounty Spectacular submission! We will review it and get back to you within a week. 🎉

Best,
Bri

**Comment by ElliotFriend** [comment]
Hi, there, @fazzatti!!

Oh, man!! This is such an incredible example! I'm way impressed with what you've put together, and how you've done it! The interactive demo is *\*chef's kiss\** simply awesome! You've done a **really** good job of implementing the standard token interface with customized, reasonable, and useful functionality. I couldn't give you more props, even if I tried!

One thing I'd say about the demo, when I try to make a transfer after I've hit one of the quotas, I see the `Transaction Failed` toast that pops up, but it'd be cool if the toasts were a bit more prominent (maybe all the toasts in general) and/or if it said something like "inflow quota exceeded" or something. That's probably just more a personal preference thing, anyway.

Overall, amazing submission! Thanks for sharing it!

**Reply by ElliotFriend** [reply]
Just for my own curiosity, do you have a public repo for your demo site? I'd love to poke around how you built the frontend. It looks really nice!

**Reply by fazzatti** [reply]
Hi @ElliotFriend, I'm so glad you liked it!!! ❤️ 

Thank you so much for the feedback! Indeed this is the portion I want to cover in the next video, going over all errors and events to display meaningful information for the end user. 

About the frontend, it is in another repo alongside the website, that's why I didn't release it. But it is a great idea that I didn't think of initially, I'll definitely look into adding a simple version of it along with the soroban demo as a reference. Probably will tackle these two in the next few weekends 🙏 

Thanks again!!

**Comment by briwylde08** [comment]
Hi @fazzatti! We have reviewed your submission. [🎉](https://emojipedia.org/party-popper/)

Given our guidelines, your submission may qualify for an award of $1,500 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

Thanks again! Please let us know if you have any questions.

Best,
Bri

**Reply by fazzatti** [reply]
Thank youuu Bri! I'll do it right away!!

**Comment by briwylde08** [comment]
Thank you for providing the requested information! You’ve passed KYC and we’ve logged your award payment of $1,500 worth of XLM, which will be sent to the wallet you designated in the next week.

Thanks again and please let me know if you have any questions!

Best,
Bri

**Reply by fazzatti** [reply]
thank you so much!!
